README.md

```diff
@@ -45,23 +46,76 @@ The webhook application can be configured using the flags outlined below.
---rule-ingress-collision                         Whether ingress tls and host collision should be checked 
---rule-ingress-violation-message                 Additional message to be included whenever any of the ingress-related rules are violated.
---tls-cert-file string                           Path to the certificate file. Required, unless --no-tls is set.
---tls-private-key-file string                    Path to the certificate key file. Required, unless --no-tls is set.
+--listen-port int32                              		Port to listen on. (default 443)
+--no-tls                                         		Do not use TLS.
+--rule-resource-limit-cpu-must-be-nonzero        		Whether 'cpu' limit in resource specifications must be a nonzero value.
+--rule-resource-limit-cpu-required               		Whether 'cpu' limit in resource specifications is required.
+--rule-resource-limit-memory-must-be-nonzero     		Whether 'memory' limit in resource specifications must be a nonzero value.
+--rule-resource-limit-memory-required            		Whether 'memory' limit in resource specifications is required.
+--rule-resource-request-cpu-must-be-nonzero      		Whether 'cpu' request in resource specifications must be a nonzero value.
+--rule-resource-request-cpu-required             		Whether 'cpu' request in resource specifications is required.
+--rule-resource-request-memory-must-be-nonzero   		Whether 'memory' request in resource specifications must be a nonzero value.
+--rule-resource-request-memory-required          		Whether 'memory' request in resource specifications is required.
+--rule-security-readonly-root-filesystem-required		Whether 'readOnlyRootFilesystem' in security context specifications is required.
+--rule-resource-violation-message                		Additional message to be included whenever any of the resource-related rules are violated.
+--rule-ingress-collision                         		Whether ingress tls and host collision should be checked 
+--rule-ingress-violation-message                 		Additional message to be included whenever any of the ingress-related rules are violated.
+--tls-cert-file string                           		Path to the certificate file. Required, unless --no-tls is set.
+--tls-private-key-file string                    		Path to the certificate key file. Required, unless --no-tls is set.
+--admission-writable-root-required-annotations-prefix	What annotation prefix should be used for readonly root filesystem check exceptions.
 ```
 
+In case you want to check `readOnlyRootFilesystem` property globally but also allow some containers requiring writable root filesystem, they can be whitelisted by using annotations.
+Annotations are required at Pod level and have default prefix `exceptions.rorootfs.validation` which can be changed by configuration flag.
+
+There are two ways how to whitelist container:
+* Using annotation per container in format `{annotation_prefix}/{container_name}: true`
```

Comment:
From my point of view, the proposed annotation design has several shortcomings.

1. I don't particularly like the idea of annotation keys containing _data_ (pod names in this case). Annotation keys are meant to be "fixed".
2. I'd prefer the default annotation prefix to be more general: cover the whole admission webhook project and adhere to [standard naming](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set): i.e. something like `admission.avast.com`
3. Why have two separate syntaxes for whilelisting containers? I don't think it adds much value.

Having said the above, I'd prefer a much simpler mechanism of a single fixed annotation along the lines of:
`admission.avast.com/readonly-rootfs.exception.containers: "container-name-1,container-name-2"`

Few things to note about what I'm proposing:
- List of container names is comma-separated. This seems to me to be much simpler and more intuitive than the JSON-like syntax such as `'["{container_name_1}","{container_name_2}"]'`.
- The _name segment_ of the annotation key (i.e. after the prefix) is `kebab-case`. This is how all built-in (and also most third party) annotations are usually named.

Reply:
I like `admission-validation.luft.avast.com/readonly-rootfs.exception.containers`.
I want to discuss the value notation in person.

Reply:
@mighq Please, summarize your thoughts here ;)

Reply:
Regarding the default annotation prefix, the keyword here is _default_. No defaults should contain any references to Luft, since it is an internal Avast project.

Reply:
You got me, let's use comma separated list (nginx ingress controller has it).
I am OK with the kebab-case.
Use empty string prefix as default annotation prefix, so we don't have any question about open-source variant. That will mean that no part before slash is expected (and slash will be ommited).

Reply:
Only support one way without container name in the annotation key.

Reply:
After discussions, I will make following changes:

- There will be only one annotation for whitelisting - with value containing list of containers
- Default annotation will not have prefix, but it might be added by config parameter
- Annotation name will be `readonly-rootfs-containers-whitelist`
- Accepted value for annotation will be comma-separated list in one string`"cont-name-1, cont-name-2"`
- There will be new config introduced to enable/disable annotation whitelisting (to enable "softer" enforcing of new policies)